crawler/filtered_positive_identifier.py

```diff
@@ -14,23 +15,20 @@
 from crawler.types import Sample
 
 
-class FilteredPositiveIdentifier:
-    # record/reference all versions and definitions here
-    versions = [
-        "v1",  # initial implementation, as per GPL-669
-    ]
-    result_regex = re.compile(f"^{POSITIVE_RESULT_VALUE}", re.IGNORECASE)
-    root_sample_id_control_regex = re.compile("^CBIQA_")
-    ct_value_limit = decimal.Decimal(30)
-    d128_context = create_decimal128_context()
+# record/reference all versions and definitions here
+FILTERED_POSITIVE_VERSION_0 = "v0"  # pre-filtered_positive definitions
+FILTERED_POSITIVE_VERSION_1 = "v1"  # initial implementation, as per GPL-669
+FILTERED_POSITIVE_VERSION_2 = "v2"  # updated as per GPL-699 and GPL-740
 
-    def current_version(self) -> str:
-        """Returns the current version of the identifier.
 
-        Returns:
-            {str} - the version number
-        """
-        return self.versions[-1]
+class FilteredPositiveIdentifier(ABC):
+    def __init__(self):
+        self.version = None
+        self.ct_value_limit = decimal.Decimal(30)
```

Comment:
Doesn't it need ```create_decimal128_context``` to create this decimal?

Reply:
Hmmm not sure. I have made the change to define within a context, and the tests still pass, so will go with that

Reply:
Actually, looking at what we do I don't think so. This value is just a `Decimal`. Then later on we, in the `d128_context` covert `Decimal128` values to `Decimal` to allow comparing. I don't think defining the ct value limit `Decimal` need be done in the context